src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,113 @@
+import { store } from '../store.js'
+
+import {
+  ROOT_TOKEN,
+  RANKED_ROOT
+} from '../constants.js'
+// util
+import {
+  head,
+  contextOf,
+  unroot,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  getSortPreference,
+  isFunction,
+  meta,
+  pathToContext,
+  getContextsSortedAndRanked,
+  hashThought,
+  splitChain,
+  nextSibling as thoughtNextSibling
+} from '../util.js'
+import { isContextViewActive } from './isContextViewActive.js'
+
+const contextToThought = (context, thoughtIndex) => context.map(contextValue => {
+  const thought = thoughtIndex[hashThought(contextValue)]
+  return { value: thought.value, rank: thought.contexts[0] ? thought.contexts[0].rank : 0 }
+})
+
+const mapContextsToThoughts = (contexts, thoughtIndex) => contexts.map(({ context }) => contextToThought(context, thoughtIndex))
+
+const getContextSiblings = (context, thoughtIndex) => {
+  const contexts = getContextsSortedAndRanked(head(context))
+  return mapContextsToThoughts(contexts, thoughtIndex)
+}
+
+const nextSiblingContext = (value, context, thoughtIndex) => {
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  const currentIndex = contextSiblings.findIndex(thoughts => thoughts.find(thought => thought.value === value))
```

Comment:
If you ignore `rank`, this might not work with duplicate thoughts (thoughts with the same value and different ranks within the same context).

Reply:
`equalThoughtRanked` does the full comparison of both `value` and `rank`.

Reply:
I've noticed some ambiguities in the rank information. I will look further into that and update the code accordingly.